scripts/eurostat/health_determinants/alcohol_consumption/process.py

```diff
@@ -0,0 +1,820 @@
+                                                      ip6=education,
+                                                      ip7=incomequin,
+                                                      ip8=residence,
+                                                      ip9=countryofbirth,
+                                                      ip10=citizenship) + "\n"
+
+        # Writing Genereated MCF to local path.
+        with open(self.mcf_file_path, 'w+', encoding='utf-8') as f_out:
+            f_out.write(final_mcf_template.rstrip('\n'))
+        # pylint: enable=R0914
+        # pylint: enable=R0912
+        # pylint: enable=R0915
+
+    def process(self):
+        """
+        This Method calls the required methods to generate
+        cleaned CSV, MCF, and TMCF file.
+        Args:
+            None
+        Returns:
+            None
+        """
+
+        final_df = pd.DataFrame(columns=['time','geo','SV','observation',\
+            'Measurement_Method'])
+        # Creating Output Directory
+        output_path = os.path.dirname(self.cleaned_csv_file_path)
+        if not os.path.exists(output_path):
+            os.mkdir(output_path)
+        sv_list = []
+
+        for file_path in self.input_files:
+            df = pd.read_csv(file_path, sep='\t', header=0)
+            file_name = file_path.split("/")[-1][:-4]
+            function_dict = {
+                "hlth_ehis_al1b": alcoholconsumption_by_sex_country_of_birth,
+                "hlth_ehis_al1c": alcoholconsumption_by_sex_citizen,
+                "hlth_ehis_al1e": alcoholconsumption_by_sex_education,
+                "hlth_ehis_al1i": alcoholconsumption_by_sex_income,
+                "hlth_ehis_al1u": alcoholconsumption_by_sex_urbanisation,
+                "hlth_ehis_al2e": hazardousalcoholconsumption_by_sex_education,
+                "hlth_ehis_al2i": hazardousalcoholconsumption_by_sex_income,
+                "hlth_ehis_al2u": hazardousalcoholconsumption_by_sex_urbanisation,
+                "hlth_ehis_al3e": bingedrinking_by_sex_education,
+                "hlth_ehis_al3i": bingedrinking_by_sex_income,
+                "hlth_ehis_al3u": bingedrinking_by_sex_urbanisation,
+                "hlth_ehis_de6": historical_bingedrinking_by_sex_education,
+                "hlth_ehis_de10": historical_alcoholconsumption_by_sex_education
+            }
+            df = function_dict[file_name](df)
+            df['SV'] = df['SV'].str.replace('_Total', '')
+            final_df = pd.concat([final_df, df])
+            sv_list += df["SV"].to_list()
+
+        final_df = final_df.sort_values(by=['time', 'geo', 'SV','observation'])
+        final_df = final_df.drop_duplicates(subset=['time','geo','SV'],\
+            keep='first')
+        final_df['observation'] = final_df['observation'].astype(str)\
+            .str.strip()
+        # derived_df generated to get the year/SV/location sets 
+        # where 'u' exist
```

Comment:
Please expand this comment. (Does alchohol dataset has only 'u'; I thought it has other cases like 'd' as well.)

Reply:
Expanded the comment and covered d case as well.